src/status/TransactionClaimed.tsx

```diff
@@ -23,12 +26,28 @@ const Broadcasting = () => {
 const TransactionClaimed = () => {
     const navigate = useNavigate();
     const { swap } = usePayContext();
-    const { t, denomination, separator } = useGlobalContext();
+    const { t, denomination, separator, setSwapStorage } = useGlobalContext();
 
     const [claimBroadcast, setClaimBroadcast] = createSignal<
         boolean | undefined
     >(undefined);
 
+    const [preimage] = createResource(async () => {
+        const submarine = swap() as SubmarineSwap;
+        if (submarine?.type !== SwapType.Submarine) {
+            return undefined;
+        }
+
+        if (submarine.preimage !== undefined) {
+            return submarine.preimage;
+        }
+
+        const res = await getSubmarinePreimage(submarine.id);
+        submarine.preimage = res.preimage;
+        await setSwapStorage(submarine);
```

Comment:
We could validate the preimage here aswell, but boltz doesnt really have any incentive to lie about it so ig its fine.

Reply:
Still, you are right. Let's do a check